cljfmt/src/cljfmt/core.cljc

```diff
@@ -341,6 +341,28 @@
   ([form indents alias-map]
    (transform form edit-all should-indent? #(indent-line % indents alias-map))))
 
+(defn map-key? [zloc]
+  (and (z/map? (z/up zloc)) (even? (index-of zloc)) (not (whitespace-or-comment? zloc))))
+
+(defn preceded-by-newline? [zloc]
+  (loop [previous (zip/left zloc)]
+    (cond
+      (nil? previous)
+      false
+      (line-break? previous)
+      true
+      (whitespace-or-comment? previous)
+      (recur (zip/left previous))
+      :else
+      false)))
+
+(defn map-key-without-line-break? [zloc]
+  (and (map-key? zloc) (zip/left zloc) (not (preceded-by-newline? zloc))))
```

Comment:
Why have `(zip/left zloc)` here? Isn't that covered by `(not (preceded-by-newline? zloc))`?